### PR TITLE
fix(CMF): remove immutable elements in collections

### PR DESCRIPTION
**Pull request by jsomsanith**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
The collection deletion tests `element.id` but the collection with immutable elements fails.

**What is the new behavior?**
We now test if element.id is defined, if not it is considered as immutable

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration
path for existing applications: ...

**Other information**:

### CODE REVIEW

File: packages/cmf/src/reducers/collectionsReducers.js [hunk 1]
```diff
@@ -7,6 +7,17 @@ import ACTIONS from '../actions';
 export const defaultState = new Map();
 
 /**
+ * Get element id. If it doesn't have "id" property, we consider it as immutable.
+ */
+function getId(element) {
```
Comment (jmfrancois) [inline packages/cmf/src/reducers/collectionsReducers.js]: add export + unittest

File: packages/cmf/src/reducers/collectionsReducers.js [hunk 2]
```diff
@@ -7,6 +7,17 @@ import ACTIONS from '../actions';
...
+	const id = element.id;
+	if (id === undefined) {
+		return element.get('id');
+	}
+	return id;
```
Comment (frassinier) [inline packages/cmf/src/reducers/collectionsReducers.js]: Does it work? 
```javascript
function getId(element) {
  return element.id || element.get('id');
}
```

Comment (jsomsanith) [inline packages/cmf/src/reducers/collectionsReducers.js]: no, if `element.id === 0`

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://282.talend.surge.sh)

File: packages/cmf/src/reducers/collectionsReducers.js (hunk 1 above)
Comment (jsomsanith) [inline packages/cmf/src/reducers/collectionsReducers.js]: Done

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://282.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://282.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://282.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://282.talend.surge.sh)

**Pull request merged**